helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -171,13 +168,6 @@
   private long _continuousResourceRebalanceFailureCount = 0;
   private long _continuousTaskRebalanceFailureCount = 0;
 
-  /**
-   * The _paused flag is checked by function handleEvent(), while if the flag is set handleEvent()
-   * will be no-op. Other event handling logic keeps the same when the flag is set.
-   */
-  private boolean _paused;
-  private boolean _inMaintenanceMode;
```

Comment:
I would suggest you to have the PR more self contained. If you moved these two into cache, how we guarantee the logic of the checks these variables? Better to have the change of checks with new cache functions together.